R/simpleheatmap3.R

```diff
@@ -45,17 +46,23 @@ simpleheatmap3 <- function(pln,
                            margins=c(5,5),scale="none",
                            plot = TRUE,
                            nrOfClustersCol = 3,
-                           nrOfClustersRow = 3, ...)
+                           nrOfClustersRow = 3,
+                           suppressColSideCols = FALSE, ...)
 {
-  if(plot) {
-    tmp <- heatmap3::heatmap3( as.matrix(pln) , scale=scale  , col=palette ,
+  if(plot&!suppressColSideCols) {
+    tmp0 <- cutree(hclustf(distf(t(as.matrix(pln)))), nrOfClustersCol)
+    colsidecolors <- rainbow(nrOfClustersCol)[tmp0]
+    tmp <- heatmap3::heatmap3(as.matrix(pln), scale=scale, col=palette,
               labRow=labRow,
               labCol = labCol,
               cexRow=0.1 + 1/log10(dim(pln)[1]),
               cexCol=0.1 + 1/log10(dim(pln)[2]),
               distfun=distf,hclustfun=hclustf,
               margins=margins,main=main,
               keep.dendro = TRUE,
+              ColSideColors = colsidecolors,
```

Comment:
instead of else if below just
```suggestion
              ColSideColors = ifelse(suppressColSideCols, NULL, colsidecolors),
```